src/builder.rs

```diff
@@ -127,95 +126,107 @@ impl BuilderFactory {
-    ///
-    /// Note that this causes the GTK widgets to be created (but not to be shown or be connected to
-    /// anything)
-    pub fn build(&self) -> BuilderUtilizer<A, W, S> {
-        Builder::from_string(&self.xml).into()
+impl <'a, A> std::ops::Deref for ActorBuilderContext<'a, A>
+where
+    A: actix::Actor<Context = actix::Context<A>>
+{
+    type Target = actix::Context<A>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.ctx
+    }
+}
+
+impl <'a, A> std::ops::DerefMut for ActorBuilderContext<'a, A>
+where
+    A: actix::Actor<Context = actix::Context<A>>
+{
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.ctx
     }
 }
 
 /// Context for utilizing a `gtk::Builder` and connecting it to he Actix world.
+/// 
+/// It wraps a `gtk::Builder` instance and provides methods to create actors that are
+/// connected to the widgets in that builder.
 ///
 /// See [`woab::Factory`](struct.Factory.html) for usage example.
-pub struct BuilderUtilizer<A, W, S> {
+pub struct BuilderConnector {
     builder: gtk::Builder,
-    _phantom: std::marker::PhantomData<(A, W, S)>,
+    callbacks: HashMap<String, RawSignalCallback>,
 }
 
-impl<A, W, S> From<gtk::Builder> for BuilderUtilizer<A, W, S> {
+impl From<gtk::Builder> for BuilderConnector {
     fn from(builder: gtk::Builder) -> Self {
         Self {
             builder,
-            _phantom: Default::default(),
+            callbacks: HashMap::new(),
         }
     }
 }
 
-impl<A, W, S> BuilderUtilizer<A, W, S>
-where
-    for<'a> &'a gtk::Builder: TryInto<W>
-{
+impl BuilderConnector {
     /// Create a widgets struct (as defined by the `W` generic parameter of
     /// [`woab::Factory`](struct.Factory.html)) and map the builder's widgets to its fields.
-    pub fn widgets(&self) -> Result<W, <&gtk::Builder as TryInto<W>>::Error>  {
-        (&self.builder).try_into()
-    }
-}
-
-impl<A, W, S> BuilderUtilizer<A, W, S>
-where
-    A: actix::Actor<Context = actix::Context<A>>,
-    for<'a> &'a gtk::Builder: TryInto<W>,
-    S: BuilderSignal,
-    A: actix::StreamHandler<S>
-{
-    /// Create an Actix actor and connect it to the builder's widgets and signals.
-    ///
-    /// `make_actor` is a function that receives the actor context and the widgets, and is
-    /// responsible for constructing the actor struct with the widgets inside it. It can also be
-    /// used for configuring and or showing the widgets GTK-wise (though this can also be handled
-    /// by the actor afterwards)
-    pub fn actor(&self, make_actor: impl FnOnce(&mut A::Context, W) -> A) -> Result<actix::Addr<A>, <&gtk::Builder as TryInto<W>>::Error> {
-        let widgets: W = self.widgets()?;
-        Ok(<A as actix::Actor>::create(move |ctx| {
-            S::connect_builder_signals::<A>(ctx, &self.builder);
-            make_actor(ctx, widgets)
-        }))
+    pub fn connect_widgets<W>(&self) -> Result<W, <gtk::Builder as TryInto<W>>::Error> 
```

Comment:
This method's name should just be `widgets`. We are not really connecting anything here.

Reply:
We are creating references to an instantiated builder and putting them into a new struct. It's not totally "connecting", but I don't know a better word and I liked the symmetry to the signals methods.

It may be a left-over from my Java time, but I really dislike method names that don't have a verb in them (except maybe if they're a simple getter).